File: backend/components/metiers/vr_ar/tests_centralises/views/router/test_router.py
# test___init__.py – Test ultra avancé, clé en main


def test_import_router():
    assert True  # Import réussi, module présent


def test___init___structure():
    import vr_ar.views.router

    assert hasattr(vr_ar.views.router, "__doc__")


def test_list_vr_ars():
    from fastapi.testclient import TestClient
    from backend.components.metiers.vr_ar.views.router.router import router
    from fastapi import FastAPI

    app = FastAPI()
    app.include_router(router)
    client = TestClient(app)
    res = client.get("/vr_ar")
    assert res.status_code == 200
    assert res.json() == {"vr_ars": [], "total": 0}


def test_create_vr_ar_defaults():
    from fastapi.testclient import TestClient
    from backend.components.metiers.vr_ar.views.router.router import router
    from fastapi import FastAPI

    app = FastAPI()
    app.include_router(router)
    client = TestClient(app)
    res = client.post("/vr_ar", json={})
    assert res.status_code == 201
    assert res.json() == {"nom": "", "description": "", "type": "objet"}


def test_create_vr_ar_with_data():
    from fastapi.testclient import TestClient
    from backend.components.metiers.vr_ar.views.router.router import router
    from fastapi import FastAPI

    app = FastAPI()
    app.include_router(router)
    client = TestClient(app)
    data = {"nom": "Cube", "description": "Test", "type": "mesh"}
    res = client.post("/vr_ar", json=data)
    assert res.status_code == 201
    assert res.json() == data


def test_create_vr_ar_rgpd():
    from fastapi.testclient import TestClient
    from backend.components.metiers.vr_ar.views.router.router import router
    from fastapi import FastAPI

    app = FastAPI()
    app.include_router(router)
    client = TestClient(app)
    data = {"nom": "Cube", "secret": "should-not-leak"}
    res = client.post("/vr_ar", json=data)
    assert "secret" not in res.json()


def test_accessibilite_json():
    from fastapi.testclient import TestClient
    from backend.components.metiers.vr_ar.views.router.router import router
    from fastapi import FastAPI

    app = FastAPI()
    app.include_router(router)
    client = TestClient(app)
    res = client.get("/vr_ar")
    assert res.headers["content-type"].startswith("application/json")

File: backend/components/metiers/vr_ar/tests_centralises/views/router/test_router.js
// test_router.js
// Tests avancés pour le module views/router (centralisation tests)
const request = require('supertest');
const express = require('express');
const router = require('../../../views/router/router');

describe('Views Router Central Tests', () => {
    let app;
    beforeAll(() => {
        app = express();
        app.use(express.json());
        app.use(router);
    });

    it('GET /vr_ar retourne une liste vide', async () => {
        const res = await request(app).get('/vr_ar');
        expect(res.status).toBe(200);
        expect(res.body).toEqual({ d3s: [], total: 0 });
    });

    it('POST /vr_ar crée un objet vr_ar avec valeurs par défaut', async () => {
        const res = await request(app).post('/vr_ar').send({});
        expect(res.status).toBe(201);
        expect(res.body).toEqual({ nom: '', description: '', type: 'objet' });
    });

    it('POST /vr_ar crée un objet vr_ar avec données fournies', async () => {
        const data = { nom: 'Cube', description: 'Test', type: 'mesh' };
        const res = await request(app).post('/vr_ar').send(data);
        expect(res.status).toBe(201);
        expect(res.body).toEqual(data);
    });

    it('POST /vr_ar RGPD: ne retourne pas de données sensibles', async () => {
        const data = { nom: 'Cube', secret: 'should-not-leak' };
        const res = await request(app).post('/vr_ar').send(data);
        expect(res.body).not.toHaveProperty('secret');
    });

    it('Accessibilité: toutes les routes répondent en JSON', async () => {
        const res = await request(app).get('/vr_ar');
        expect(res.headers['content-type']).toMatch(/application\/json/);
    });
});
